src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl

```diff
@@ -184,7 +242,7 @@ KERNEL(fc)(
                         const uint total_k = ki * TILE_K + kii;
                         if (total_k < LEFTOVER_IFM) {
                             INPUT0_TYPE in_val = _sub_group_shuffle(((INPUT0_TYPE*)(&in_0[bi]))[total_k / SIMD], total_k % SIMD);
-                            ((ACCUMULATOR_TYPE*)(&acc[bi]))[fi] += in_val * ((FILTER_TYPE*)(&wei))[kii * TILE_OFM + fi];
+                            ((ACCUMULATOR_TYPE*)(&acc[bi]))[fi] += in_val * ((ACCUMULATOR_TYPE*)(&wei))[kii * TILE_OFM + fi];
```

Comment:
not exactly sure about current behavior, but for fp16 computation, I expect FILTER_TYPE to be fp16 and ACCUMULATOR_TYPE to be fp32. Otherwise, we cannot expect the MAC perf advantage of fp16. But if we cast the datatype of wei into ACCUMULATOR_TYPE, I'm afraid it may bring negative perf change.

Reply:
accumulator type is also fp16. `mad` instruction requires all operands to have half type.